perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -1461,7 +1479,7 @@ public void run() {
 				log.debug("Synchronization thread for group {} has finished in {} ms.", group, System.currentTimeMillis()-startTime);
 			} catch (WrongAttributeValueException | WrongReferenceAttributeValueException | InternalErrorException |
 					WrongAttributeAssignmentException | MemberAlreadyRemovedException | GroupNotExistsException |
-					GroupOperationsException | AttributeNotExistsException | ExtSourceNotExistsException e) {
+					AttributeNotExistsException | AlreadyMemberException | ExtSourceNotExistsException e) {
```

Comment:
Once my previous comment is resolved, you can remove ``AlreadyMemberException`` and ``MemberAlreadyRemovedException`` from catch clause.